index.js

```diff
@@ -75,7 +85,9 @@ app.on('ready', () => {
 
     rpc.on('init', () => {
       win.show();
-      if (fn) fn(win);
+      if (!fn) fn = (win) => win.rpc.emit('session add req');
+      (app.windowCallback || fn)(win);
```

Comment:
Do you mean comment in the code or an explanation here?
___
The first line (line 88) sets `createWindow` callback to `win.prc.emit('session add req')` if no callback is set: this means that by default `createWindow` will create a new session, except if a callback is passed (in our specific case the callback setted by `'open-file'` app event will fire `'open file'` rpc event, and `'session add req'` will not be fired).
___
The second line (line 89) execute `app.windowCallback` (the callback setted before `createWindow` is even defined, in our specific case in `'open-file'` app event, that fires before the `'ready'` event) or the `fn` `createWindow` callback if `app.windowCallback` is not defined. 

**`(app.windowCallback || fn)(win);`** 
is exactly the same of writing 
**`app.windowCallback ? app.windowCallback(win) : fn(win);`**
___
Hope to have clarified the code, it's not so easy to explain :)